src/main/webapp/scripts/private-room-script.js

```diff
@@ -87,69 +91,41 @@ function onYouTubeIframeAPIReady() {
 
 // Log state changes
 function onStateChange(event) {
-    var state = "undefined";
-    switch (event.data) {
-        case YT.PlayerState.UNSTARTED:
-            state = "-1";
-            break;
-        case YT.PlayerState.ENDED:
-            state = "0";
-            break;
-        case YT.PlayerState.PLAYING:
-            state = "1";
-            break;
-        case YT.PlayerState.PAUSED:
-            state = "2";
-            break;
-        case YT.PlayerState.BUFFERING:
-            state = "3";
-            break;
-        case YT.PlayerState.CUED:
-            state = "5";
-            break;
-        default:
-            state = "unknown (" + event.data + ")";
-    }
-    timeStamp = Math.round(youtubePlayer.getCurrentTime());
-    updateCurrentState(state, timeStamp);
+    var state = event.data
+    playerTimeStamp = Math.round(youtubePlayer.getCurrentTime());
+    updateCurrentState(state, playerTimeStamp);
 }
 
 // Every three seconds you listen to youtube player for any detection of change
 window.setInterval(function(){
     listenForStateChange();
-}, 1000);
+}, setWaitTime);
 
 // This function fetches the state of the private room video and
 // plays/pauses it accordingly
-function listenForStateChange(){
-    $(document).ready(function(){
-        const Url = `/sync-room?roomId=${roomId.toString()}`;
-        $.get(Url,function(data, status){
-            // Change timestamp to match group timestamp
-            if(timeStamp + 2 < data.timestamp || timeStamp - 2 > data.timestamp){
-                youtubePlayer.seekTo(data.timestamp);
-            }
-            // Change state to match group state
-            if(data.currentState === "1"){
-                playVideo();
-            } else if(data.currentState === "2") {
-                pauseVideo();
-            }
-        })
-    })
+async function listenForStateChange(){
+    let privateRoomDataPromise = await fetch('/sync-room?roomId='+roomId);
+    // fetch the json-version of the urls for all the youtube videos
+    let privateRoomData = await privateRoomDataPromise.json();
+    // Change timestamp to match group timestamp if client is not within two seconds of room
+    if(Math.abs(playerTimeStamp - privateRoomData.timestamp) >= allowedSecondsOfDiscrepancy){
+        console.log('Seeking to ' + privateRoomData.timestamp)
```

Comment:
logging? Seems ok, but make sure it's specific enough to know where it is coming from.